Source/References/StateHoldr-0.2.0.ts

```diff
@@ -334,6 +353,7 @@ module StateHoldr {
                     "valueDefault": {},
                     "storeLocally": true
                 });
+                this.collectionKeys.push(collectionKey);
```

Comment:
You should update the value of what ItemsHolder stores. Right now you're assuming that it keeps a reference to the same object, but in the future it could instead keep a shallow or deep copy.